crates/runtime/src/accelerated_table/refresh_task.rs

```diff
@@ -550,10 +546,9 @@ impl RefreshTask {
 
         let existing_records = self
             .accelerator_df(
-                self.refresh_df_context(Arc::clone(&federated_provider)),
+                &self.refresh_df_context(Arc::clone(&federated_provider)),
```

Comment:
Doesn't this change imply:
1. There are no existing records, 
2. There is no overlap between existing and the new records
3. There is no need to `filter_records` [below](https://github.com/spiceai/spiceai/pull/3697/files#diff-90f336ae367603afbc75d3643de47bae6a7f49eadbdc061aeaac5d1d70b4433eR569). 

IIRC, The need to diff current/new records is to handle timestamp jitter in the underlying systems that may have been missed in the previous append refresh.

Reply:
Not super confident on this however.

Reply:
No, was mistaken.